ratis-assembly/pom.xml

```diff
@@ -123,14 +124,42 @@
           <skipAssembly>false</skipAssembly>
           <appendAssemblyId>true</appendAssemblyId>
           <tarLongFileMode>gnu</tarLongFileMode>
-          <descriptors>
-            <descriptor>src/main/assembly/src.xml</descriptor>
-            <descriptor>src/main/assembly/bin.xml</descriptor>
-            <descriptor>src/main/assembly/examples-bin.xml</descriptor>
-            <descriptor>src/main/assembly/shell-bin.xml</descriptor>
-          </descriptors>
         </configuration>
+        <executions>
+          <execution>
+            <id>src</id>
+            <phase>package</phase>
+            <goals>
+              <goal>single</goal>
+            </goals>
+            <configuration>
+              <descriptors>
+                <descriptor>src/main/assembly/src.xml</descriptor>
+              </descriptors>
+              <finalName>apache-ratis-${project.version}-src</finalName>
+              <appendAssemblyId>false</appendAssemblyId>
+            </configuration>
+          </execution>
+          <execution>
+            <id>default-cli</id>
```

Comment:
Let's use `bin` instead of `default-cli`.

Reply:
Hi @szetszwo, I had tried use `bin`, the build command will be failed.
```
./dev-support/checks/build.sh -Prelease assembly:single
```
```
[INFO] --- maven-assembly-plugin:3.4.1:single (default-cli) @ ratis-assembly ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.224 s
[INFO] Finished at: 2022-08-09T13:49:16+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.4.1:single (default-cli) on project ratis-assembly: Error reading assemblies: No assembly descriptors found. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:

```

Reply:
@leo65535 , thanks for trying it.  I did not know that `default-cli` is a special id in maven so that we cannot change simply change it; see https://maven.apache.org/guides/mini/guide-default-execution-ids.html . Let's keep using `default-cli` then.